Java 25 incompatibility: CDS fails with "module system cannot be modified after it's initialized"

# Issue: Java 25 incompatibility with Micronaut Test Resources CDS (Class Data Sharing)

## Environment
- **Java**: OpenJDK 25.0.2 (Zulu)
- **Micronaut Core**: 4.10.10
- **Micronaut Gradle Plugin**: 4.6.1
- **Micronaut Test Resources Plugin**: 4.6.1 (uses micronaut-test-resources 2.10.1)
- **Gradle**: 9.3.1
- **OS**: macOS (Darwin 25.3.0)

## Problem
Micronaut Test Resources fails to start when using Java 25 due to CDS (Class Data Sharing) attempting to modify the module system after initialization.

## Error
```
java.lang.IllegalArgumentException: javax.management.NotCompliantMBeanException: com.sun.management.UnixOperatingSystemMXBean: During -Xshare:dump, module system cannot be modified after it's initialized
	at java.management/javax.management.StandardMBean.<init>(StandardMBean.java:270)
	at java.management/java.lang.management.ManagementFactory.addMXBean(ManagementFactory.java:882)
	...
Caused by: java.lang.UnsupportedOperationException: During -Xshare:dump, module system cannot be modified after it's initialized
	at java.base/java.lang.Module.defineModule0(Native Method)
	at java.base/java.lang.Module.<init>(Module.java:140)
	at java.base/java.lang.reflect.Proxy$ProxyBuilder.getDynamicModule(Proxy.java:823)
	...
```

## Root Cause
The test resources service process is launched with `-Xshare:dump` flag to create a CDS archive. Java 25's stricter module system doesn't allow module modifications during CDS dump, specifically when JMX tries to create dynamic proxy modules for MXBeans.

## Reproduction
```gradle
// build.gradle
plugins {
    id("io.micronaut.application") version "4.6.1"
    id("io.micronaut.test-resources") version "4.6.1"
}

micronaut {
    version "4.10.10"
    testResources {
        additionalModules.add("kafka")
    }
}
```

```java
// Test class
@MicronautTest
class KafkaE2ETest {
    // Test resources should auto-provision Kafka
}
```

**Result**: Task `:internalStartTestResourcesService` fails with the module system error.

## Attempted Workarounds
None of these worked:
- `sharedServer = false` - Still tries CDS
- `inferClasspath = false` - No effect
- `systemProperty 'micronaut.test.resources.cds.enabled', 'false'` - No effect
- Environment variable `MICRONAUT_TEST_RESOURCES_DISABLE_CDS=true` - No effect

The plugin hardcodes the CDS flags in the JVM arguments.

## Suggested Fix
1. **Short-term**: Add a configuration option to disable CDS:
```gradle
micronaut {
    testResources {
        cds {
            enabled = false  // Or detect Java 25+ and auto-disable
        }
    }
}
```

2. **Long-term**: Make CDS compatible with Java 25's module system constraints, or auto-detect Java 25 and skip CDS.

## Current Workaround
Use explicit Testcontainers with `TestPropertyProvider`:
```java
@MicronautTest
@Testcontainers
@TestInstance(Lifecycle.PER_CLASS)
class KafkaE2ETest implements TestPropertyProvider {
    @Container
    static final KafkaContainer kafka = new KafkaContainer(...);
    
    @Override
    public Map<String, String> getProperties() {
        return Map.of("kafka.bootstrap.servers", kafka.getBootstrapServers());
    }
}
```

## Impact
This blocks Java 25 adoption for projects using Micronaut Test Resources, which is promoted as the zero-config testing solution.

--- a/test-resources-build-tools/src/main/java/io/micronaut/testresources/buildtools/ServerUtils.java
+++ b/test-resources-build-tools/src/main/java/io/micronaut/testresources/buildtools/ServerUtils.java
@@ -474,7 +474,9 @@ public class ServerUtils {
         @Override
         public Map<String, String> getSystemProperties() {
             Map<String, String> systemProperties = new HashMap<>();
-            systemProperties.put(JMX_SYSTEM_PROPERTY, null);
+            if (!isCDSDumpInvocation()) {
+                systemProperties.put(JMX_SYSTEM_PROPERTY, null);
+            }
             String dockerCheckTimeout = System.getProperty(DOCKER_CHECK_TIMEOUT_SECONDS_PROPERTY);
             if (dockerCheckTimeout == null) {
                 dockerCheckTimeout = System.getenv(DOCKER_CHECK_TIMEOUT_SECONDS_ENV);

--- a/test-resources-build-tools/src/test/groovy/io/micronaut/testresources/buildtools/ServerUtilsTest.groovy
+++ b/test-resources-build-tools/src/test/groovy/io/micronaut/testresources/buildtools/ServerUtilsTest.groovy
@@ -60,7 +60,7 @@ class ServerUtilsTest extends Specification {
         1 * factory.startServer(_) >> { ServerUtils.ProcessParameters params ->
             assert params.mainClass == 'io.micronaut.testresources.server.TestResourcesService'
             assert params.classpath == classpath
-            def sysProps = ['com.sun.management.jmxremote': null]
+            def sysProps = [(JMX_PROPERTY): null]
             if (token != null) {
                 sysProps["server.access-token"] = token
             }
@@ -197,7 +197,8 @@ class ServerUtilsTest extends Specification {
             def jvmArgs = params.jvmArguments
             assert jvmArgs.contains("-Xshare:off")
             assert jvmArgs.contains(cdsClassListOption)
-            assert params.systemProperties == [(JMX_PROPERTY): null]
+            assert params.systemProperties.containsKey(JMX_PROPERTY)
+            assert params.systemProperties.get(JMX_PROPERTY) == null
             assert params.classpath.contains(cdsFlatJar.toFile())
             assert Files.exists(cdsFlatJar)
             Files.write(cdsClassList, "test".getBytes())
@@ -224,7 +225,8 @@ class ServerUtilsTest extends Specification {
             assert !jvmArgs.contains("-Xshare:dump")
             assert !jvmArgs.contains(cdsSharedClassListOption)
             assert jvmArgs.contains(cdsSharedArchiveFileOption)
-            assert params.systemProperties == [(JMX_PROPERTY): null]
+            assert params.systemProperties.containsKey(JMX_PROPERTY)
+            assert params.systemProperties.get(JMX_PROPERTY) == null
         }
         1 * factory.waitFor(_) >> {
             portFile.toFile().text = "${embeddedServer.port}"
@@ -241,7 +243,8 @@ class ServerUtilsTest extends Specification {
             assert !jvmArgs.contains("-Xshare:dump")
             assert !jvmArgs.contains(cdsSharedClassListOption)
             assert jvmArgs.contains(cdsSharedArchiveFileOption)
-            assert params.systemProperties == [(JMX_PROPERTY): null]
+            assert params.systemProperties.containsKey(JMX_PROPERTY)
+            assert params.systemProperties.get(JMX_PROPERTY) == null
         }
         1 * factory.waitFor(_) >> {
             portFile.toFile().text = "${embeddedServer.port}"
@@ -257,7 +260,8 @@ class ServerUtilsTest extends Specification {
             def jvmArgs = params.jvmArguments
             assert jvmArgs.contains("-Xshare:off")
             assert jvmArgs.contains(cdsClassListOption)
-            assert params.systemProperties == [(JMX_PROPERTY): null]
+            assert params.systemProperties.containsKey(JMX_PROPERTY)
+            assert params.systemProperties.get(JMX_PROPERTY) == null
             assert params.classpath == []
             Files.write(cdsClassList, "test".getBytes())
         }

--- a/test-resources-build-tools/src/test/groovy/io/micronaut/testresources/buildtools/ServerUtilsTest.groovy
+++ b/test-resources-build-tools/src/test/groovy/io/micronaut/testresources/buildtools/ServerUtilsTest.groovy
@@ -14,6 +14,8 @@ import java.nio.file.Path
 import java.util.concurrent.atomic.AtomicInteger
 
 class ServerUtilsTest extends Specification {
+    private static final String JMX_PROPERTY = 'com.sun.management.jmxremote'
+
     @TempDir
     Path tmpDir
 
@@ -195,6 +197,7 @@ class ServerUtilsTest extends Specification {
             def jvmArgs = params.jvmArguments
             assert jvmArgs.contains("-Xshare:off")
             assert jvmArgs.contains(cdsClassListOption)
+            assert params.systemProperties == [(JMX_PROPERTY): null]
             assert params.classpath.contains(cdsFlatJar.toFile())
             assert Files.exists(cdsFlatJar)
             Files.write(cdsClassList, "test".getBytes())
@@ -213,6 +216,7 @@ class ServerUtilsTest extends Specification {
             assert jvmArgs.contains("-Xshare:dump")
             assert jvmArgs.contains(cdsSharedClassListOption)
             assert jvmArgs.contains(cdsSharedArchiveFileOption)
+            assert !params.systemProperties.containsKey(JMX_PROPERTY)
             Files.write(cdsArchiveFile, "test".getBytes())
         }
         1 * factory.startServer(_) >> { ServerUtils.ProcessParameters params ->
@@ -220,6 +224,7 @@ class ServerUtilsTest extends Specification {
             assert !jvmArgs.contains("-Xshare:dump")
             assert !jvmArgs.contains(cdsSharedClassListOption)
             assert jvmArgs.contains(cdsSharedArchiveFileOption)
+            assert params.systemProperties == [(JMX_PROPERTY): null]
         }
         1 * factory.waitFor(_) >> {
             portFile.toFile().text = "${embeddedServer.port}"
@@ -236,6 +241,7 @@ class ServerUtilsTest extends Specification {
             assert !jvmArgs.contains("-Xshare:dump")
             assert !jvmArgs.contains(cdsSharedClassListOption)
             assert jvmArgs.contains(cdsSharedArchiveFileOption)
+            assert params.systemProperties == [(JMX_PROPERTY): null]
         }
         1 * factory.waitFor(_) >> {
             portFile.toFile().text = "${embeddedServer.port}"
@@ -251,6 +257,7 @@ class ServerUtilsTest extends Specification {
             def jvmArgs = params.jvmArguments
             assert jvmArgs.contains("-Xshare:off")
             assert jvmArgs.contains(cdsClassListOption)
+            assert params.systemProperties == [(JMX_PROPERTY): null]
             assert params.classpath == []
             Files.write(cdsClassList, "test".getBytes())
         }

